src/main/java/uk/gov/hmcts/divorce/noticeofchange/service/ChangeOfRepresentativeService.java

```diff
@@ -0,0 +1,145 @@
+                        beforeSolicitor.getOrganisationPolicy().getOrganisation());
+            }
+            updateOrgPolicyAndSolicitorDetails(currentSolicitor, nocRequestingUser, loggedInUserEmail);
+            final ProfessionalUser professionalUser = getProfessionalUsers(sysUserToken, s2sToken, organisationId, loggedInUserEmail);
+            updatedBy = String.join(" ", professionalUser.getFirstName(), professionalUser.getLastName());
+            addedRepresentative = updateRepresentative(updatedBy, loggedInUserEmail, orgToAdd);
+        } else {
+            var userDetails = idamService.retrieveUser(request.getHeader(AUTHORIZATION)).getUserDetails();
+            var beforeSolicitor = isApplicant1 ? beforeCaseData.getApplicant1().getSolicitor()
+                    : beforeCaseData.getApplicant2().getSolicitor();
+            updatedBy = userDetails.getName();
+            addedRepresentative = updateRepresentative(currentSolicitor.getName(), currentSolicitor.getEmail(),
+                    currentSolicitor.getOrganisationPolicy().getOrganisation());
+            if (beforeSolicitor != null && beforeSolicitor.getOrganisationPolicy() != null) {
+                removedRepresentative = updateRepresentative(beforeSolicitor.getName(), beforeSolicitor.getEmail(),
+                        beforeSolicitor.getOrganisationPolicy().getOrganisation());
+            }
+        }
+
+        updateChangeOfRepresentativeTab(caseData, clientName, updatedBy,
+                updatedVia, addedRepresentative,
+                removedRepresentative, getParty(isApplicant1, caseData.getApplicationType()));
+    }
+
+    private String getParty(boolean isApplicant1, ApplicationType applicationType) {
+        if (isApplicant1) {
+            return "Applicant";
+        } else {
+            return SOLE_APPLICATION.equals(applicationType) ? "Respondent" : "Applicant2";
+        }
+    }
+
+    private void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor,
+                                                    ProfessionalUser nocRequestingUser, String loggedInUserEmail) {
+
+        applicantSolicitor.setName(String.join(" ", nocRequestingUser.getFirstName(), nocRequestingUser.getLastName()));
+        applicantSolicitor.setEmail(loggedInUserEmail);
+    }
+
+    private ProfessionalUser getProfessionalUsers(String sysUserToken, String s2sToken, String organisationId, String loggedInUserEmail) {
+        var organisationUsers = Optional.ofNullable(organisationClient.getOrganisationUsers(sysUserToken, s2sToken, organisationId))
+                .map(FindUsersByOrganisationResponse::getUsers)
+                .orElse(Collections.emptyList());
+
+        return organisationUsers.stream()
+                .filter(user -> user.getEmail().equalsIgnoreCase(loggedInUserEmail))
+                .findFirst()
+                .orElseThrow();
+    }
+
+    private Representative updateRepresentative(String solicitorName, String solicitorEmail,
```

Comment:
Maybe simplify this to accept a single parameter consisting of the solicitor object, then get the fields needed within this method itself?

Reply:
Added a method to accept single paramater